state/state.go

```diff
@@ -27,3 +27,41 @@ type View interface {
 	NewView(ctx context.Context, changes merkledb.ViewChanges) (merkledb.View, error)
 	GetMerkleRoot(ctx context.Context) (ids.ID, error)
 }
+
+type Key struct {
+	Name string
+	Mode [ModeLen]byte
+}
+
+const (
+	ModeLen = 8
+	Read    = 0
+	Write   = 1
+)
+
+func NewKey(name string, bits ...int) Key {
+	var key Key
+	key.Name = name
+
+	for _, bit := range bits {
+		byteIdx := bit / 8
+		bitIdx := uint(bit % 8)
+
+		if byteIdx < ModeLen {
+			key.Mode[byteIdx] |= (1 << bitIdx)
+		}
+	}
+
+	return key
+}
+
+func (k *Key) HasMode(mode int) bool {
```

Comment:
We should just make `NewKey#bits` and `HasMode#mode` have the same variable name (either bit or index or mode or perm or something like that...) since they all mean the same thing of the bit to check for consistency